build/compile/Microsoft.DotNet.Cli.Archive.targets

```diff
@@ -0,0 +1,49 @@
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <Target Name="GenerateNuGetPackagesArchive"
+            DependsOnTargets="RestoreNuGetPackagesArchive;
+                              CompressNuGetPackagesArchive"
+            Inputs="$(Stage2Directory)/sdk/**/dotnet.dll"
+            Outputs="$(FinalArchive)">
+      <Message Text="Generating NuGet Packages Archive" />
+    </Target>
+
+    <Target Name="RestoreNuGetPackagesArchive"
+            DependsOnTargets="SetupNuGetPackagesArchiveInputsOutputs">
+      <ItemGroup>
+        <FilesToClean Include="$(NuGetPackagesArchiveProject)/**/*" />
+        <FilesToClean Include="$(NuGetPackagesArchiveFolder)/**/*" />
+      </ItemGroup>
+
+      <Delete Files="@(FilesToClean)" />
+      <RemoveDir Directories="$(NuGetPackagesArchiveProject);$(NuGetPackagesArchiveFolder)" />
+      <MakeDir Directories="$(NuGetPackagesArchiveProject);$(NuGetPackagesArchiveFolder)"/>
+
+      <Exec Command="$(DotnetStage2) new" WorkingDirectory="$(NuGetPackagesArchiveProject)" />
+      <Exec Command="$(DotnetStage2) restore --packages $(NuGetPackagesArchiveFolder)"
+          WorkingDirectory="$(NuGetPackagesArchiveProject)" />
+    </Target>
+
+    <Target Name="CompressNuGetPackagesArchive"
```

Comment:
Yes. The very first target has inputs/outputs set on it already. Though I just realize it won't work because I was using a property that wasn't defined yet. I did some refactoring and it will work now.